### ISSUE TITLE
expected-equal false positive

### ISSUE THREAD

**Issue by jalaziz** [Antonboom/testifylint#61]
testifylint falsely thinks the arguments to `assert.Equal` should be flipped in this example:

```
		event := &events.Event{
			EventMetadata: events.EventMetadata{
				ID:        "1",
				Topic:     "test-topic",
			},
			RawData: []byte("test-data"),
		}

		subscriber.processEvent(event)

		for _, sub := range subscriber.subscriptions["test-topic"] {
			select {
			case receivedEvent := <-sub.channel:
				assert.Equal(t, event, &receivedEvent)
			default:
				assert.Fail(t, "expected channel to receive event")
			}
		}
```

This started occurring with the recent release of `golangci-lint`. The previous release, containing an older version of testifylint did not falsely detect warn in this scenario.
